coldfront/core/statistics/views.py

```diff
@@ -0,0 +1,252 @@
+                role__name__in=['Principal Investigator', 'Manager']).exists():
+            return True
+
+        if job_obj.userid == self.request.user:
+            return True
+
+        messages.error(
+            self.request, 'You do not have permission to view the previous page.')
+        return False
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        job_obj = self.get_object()
+        context['job'] = job_obj
+
+        context['status_danger_list'] = ['NODE_FAIL',
+                                         'CANCELLED',
+                                         'FAILED',
+                                         'OUT_OF_MEMORY',
+                                         'TIMEOUT']
+
+        context['status_warning_list'] = ['PREEMPTED',
+                                          'REQUEUED']
+
+        context['nodes'] = ', '.join([x.name for x in job_obj.nodes.all()])
+
+        return context
+
+
+class ExportJobListView(LoginRequiredMixin,
+                        UserPassesTestMixin,
+                        View):
+
+    def test_func(self):
+        """ UserPassesTestMixin Tests"""
+        if self.request.user.is_superuser:
+            return True
+
+        if self.request.user.has_perm('statistics.view_job'):
+            return True
+
+    def dispatch(self, request, *args, **kwargs):
+        if self.test_func():
+            data = copy.deepcopy(self.request.session.get('job_search_form_data'))
+
+            job_list = Job.objects.all()
+
+            if data:
+                for date in ['submitdate', 'startdate', 'enddate']:
+                    if data.get(date):
+                        data[date] = datetime.strptime(data.get(date), DATE_FORMAT)
+                job_list = job_query_filtering(job_list, data)
+
+            class Echo:
+                def write(self, value):
+                    return value
+
+            echo_buffer = Echo()
+            writer = csv.writer(echo_buffer)
+            header = ('jobslurmid',
+                      'username',
+                      'project_name',
+                      'partition',
+                      'jobstatus',
+                      'submitdate',
+                      'startdate',
+                      'enddate')
+            job_list = job_list.values_list('jobslurmid',
+                                            'userid__username',
+                                            'accountid__name',
+                                            'partition',
+                                            'jobstatus',
+                                            'submitdate',
+                                            'startdate',
+                                            'enddate')
+            rows = (writer.writerow(row) for row in itertools.chain([header], job_list.iterator()))
+            response = StreamingHttpResponse(rows, content_type="text/csv")
```

Comment:
Thanks for using this.

Judging from `top` output, `httpd`'s memory usage stays low during the download, but its CPU usage stays at 95%+ of my two-core VM throughout. I'll evaluate this on the staging instance when we deploy there.